### DISCUSSION TITLE
Java API to run CodeQL queries?

### DISCUSSION THREAD

**Discussion by knutwannheden** [github/codeql#12315] [Q&A] [answered]
Is there also a Java API that can be used to run CodeQL queries against a database? I could not find the Java extractor anywhere in the repository, so I assume that there must be a non-public `codeql-cli` repository. But still, is there maybe some public Java API?

**Comment by hmakholm** [comment, marked as answer]
Indeed, neither the Java extractor nor the core QL engine are open-sourced.

Even though the major part of the CodeQL CLI happens to be a Java program (for now!), this is to be considered an implementation detail, and we don't support calling it directly from Java.
